nuxeo-features/nuxeo-automation/nuxeo-automation-core/src/main/java/org/nuxeo/ecm/automation/core/operations/document/CopyDocument.java

```diff
@@ -60,6 +68,9 @@ public DocumentModel run(DocumentRef ref) {
         if (name == null || name.length() == 0) {
             n = session.getDocument(ref).getName();
         }
+        if (Boolean.TRUE.equals(resetLifeCycle)) {
+            return session.copy(ref, target, n, CopyOption.RESET_LIFE_CYCLE);
+        }
         return session.copy(ref, target, n);
     }
 }
```

Comment:
i think we refactor the code to avoid duplication on `name` null check and the copy with life cycle, as proposition 
```
@OperationMethod(collector = DocumentModelCollector.class)
    public DocumentModel run(DocumentModel doc) {
        return makeCopy(doc.getRef(), name);
    }

    @OperationMethod(collector = DocumentModelCollector.class)
    public DocumentModel run(DocumentRef ref) {
        return makeCopy(ref, name);
    }

    public DocumentModel makeCopy(DocumentRef documentRef, String name) {
        String n = name;
        if (StringUtils.isEmpty(name@)) {
            n = session.getDocument(documentRef).getName();
        }
        CopyOption[] options = resetLifeCycle ? new CopyOption[] { CopyOption.RESET_LIFE_CYCLE } : new CopyOption[] {};
        return session.copy(documentRef, target, n, options);
    }
```

Reply:
did 
```
    
@OperationMethod(collector = DocumentModelCollector.class)
    public DocumentModel run(DocumentModel doc) {
        String n = name;
        if (name == null || name.length() == 0) {
            n = doc.getName();
        }
        List<CopyOption> options = new ArrayList<>();

        if (resetLifeCycle) {
            options.add(CopyOption.RESET_LIFE_CYCLE);
        }
        return session.copy(doc.getRef(), target, n, options.toArray(new CopyOption[0]));
    }

    @OperationMethod(collector = DocumentModelCollector.class)
    public DocumentModel run(DocumentRef ref) {
        return run(session.getDocument(ref));
    }
```

Reply:
👍 thx, you can use `options.toArray(CopyOption[]::new);` and `name == null || name.length() == 0` can be replaced by `StringUtils.isEmpty(name)` (it's a nice to have)

Reply:
Did it thanks !